### PR TITLE
Geo shape file element inspector preview

### PR DESCRIPTION
**Pull request by joeybrk372**
- [x] joeferner
- [x] mwizeman joeybrk372 jharwig sfeng88

requires https://github.com/visallo/visallo/pull/1433

CHANGELOG
Added: Imported files with geo data (.geojson, .kml) show a preview in the inspector.

### CODE REVIEW

File: web/plugins/map-product/src/main/resources/org/visallo/web/product/map/OpenLayers.jsx [hunk 1]
```diff
@@ -3,32 +3,33 @@ define([
...
     'product/toolbar/ProductToolbar'
 ], function(
     createReactClass,
     PropTypes,
     ol,
     jsonpatch,
-    MultiPointCluster,
+    layerHelpers,
     ProductToolbar) {
+    'use strict';
```
Comment (joeferner) [inline web/plugins/map-product/src/main/resources/org/visallo/web/product/map/OpenLayers.jsx]: Just curious, does this do anything now that we are running things through babel?

File: web/plugins/map-product/src/main/resources/org/visallo/web/product/map/worker/actions-impl.js [hunk 2]
```diff
@@ -76,6 +76,36 @@ define([
             })
         },
 
+        setElementData: ({ productId, elements, undoable }) => (dispatch, getState) => {
+            const state = getState();
+            const workspaceId = state.workspace.currentId;
+            let undoPayload = {};
+            if (undoable) {
+                // TODO: reenable
```
Comment (joeferner) [inline web/plugins/map-product/src/main/resources/org/visallo/web/product/map/worker/actions-impl.js]: Remove?

File: web/plugins/map-product/src/main/resources/org/visallo/web/product/map/worker/actions-impl.js [hunk 3]
```diff
@@ -76,6 +76,36 @@ define([
...
+                //};
+            }
+
+            dispatch({
+                type: 'PRODUCT_MAP_ADD_ELEMENTS',
+                payload: {
+                    workspaceId,
+                    productId,
+                    elements
+                    //...undoPayload
```
Comment (joeferner) [inline web/plugins/map-product/src/main/resources/org/visallo/web/product/map/worker/actions-impl.js]: Remove?

Comment (joeybrk372) [reply web/plugins/map-product/src/main/resources/org/visallo/web/product/map/worker/actions-impl.js]: This code is branched from https://github.com/visallo/visallo/tree/product-annotations, these changes are already fixed in https://github.com/visallo/visallo/pull/1432

File: web/plugins/map-product/src/main/resources/org/visallo/web/product/map/worker/actions-impl.js (hunk 2 above)
Comment (joeybrk372) [reply web/plugins/map-product/src/main/resources/org/visallo/web/product/map/worker/actions-impl.js]: This code is branched from https://github.com/visallo/visallo/tree/product-annotations, these changes are already fixed in https://github.com/visallo/visallo/pull/1432

File: core/core/src/main/java/org/visallo/core/config/Configuration.java [hunk 4]
```diff
@@ -336,7 +336,7 @@ public Map toMap() {
     }
 
     public Iterable<String> getKeys() {
-        return this.config.keySet();
+        return new HashSet<>(this.config.keySet());
```
[the remark this answers is not in the document]
Comment (joeybrk372) [reply core/core/src/main/java/org/visallo/core/config/Configuration.java]: deferring to @mwizeman, this change is from https://github.com/visallo/visallo/pull/1432

File: web/plugins/map-product/src/main/resources/org/visallo/web/product/map/OpenLayers.jsx (hunk 1 above)
Comment (joeybrk372) [reply web/plugins/map-product/src/main/resources/org/visallo/web/product/map/OpenLayers.jsx]: By default babel will automatically add `use strict`, so this isn't really necessary. I must have just put that out of old habit.

**Pull request merged**